src/ophys_etl/transforms/sine_dewarp.py

```diff
@@ -0,0 +1,606 @@
+                if 0 != noise_reduction:
+                    low_mask = (col < 0)
+                    col[low_mask] = 0  # underflow?
+
+                high_mask = (col > maxlevel)
+                col[high_mask] = maxlevel  # saturated?  for max image==1.0
+
+                imgout[:, 511 - j] = col
+            else:
+                imgout[:, 511 - j] = (
+                    imgin[:, 511 - xindexR[j]]
+                    * xtable['bgfactorR']
+                )
+        else:
+            imgout[:, 511 - j] = xtable['mean_modevalue'] * xtable['bgfactorR']
+
+    if FOVwidth == 512:
+        img = imgout.astype(np.uint16)
+    else:
+        img = imgout[:, xtable['L']:512 - xtable['R']].astype(np.uint16)
+
+    return img
+
+
+def get_xindex(a: float, b: float):
+    """
+    Generate the xindex arrays that will be used in the dewarping process.
+
+    Parameters
+    ----------
+    a: float
+        Information about the rig used for the experiment.
+    b: float
+        Information about the rig used for the experiment.
+    Returns
+    -------
+    np.array
+        The data ready to use in the dewarping process
+    np.array
+        The data ready to use in the dewarping process
+    """
+    xindex = np.zeros(256, np.int)
+    xindexB = np.zeros(256, np.float)  # between pixels
+
+    for j in range(0, int(a)):
+        xindex[j] = (
+            j - int(
+                    (b)
+                    * (1.0 - math.sin((j/(a * 3.0) + 1.0/6.0) * 3.14159265))
+                    + 0.5
+                )
+        )
+
+        # TODO: Should these also use int()?
+        xindexB[j] = (
+            (j + 0.5) - (
+                (b) * (1.0 - math.sin(
+                    ((j + 0.5)/(a * 3.0) + 1.0 / 6.0) * 3.14159265
+                ))
+            )
+        )
+
+    return xindex, xindexB
+
+
+def create_xtable(movie: np.ndarray,
+                  aL: float,
+                  aR: float,
+                  bL: float,
+                  bR: float,
+                  noise_reduction: int):
+    """
+    Compute a number of statistics about the images to be dewarped.
+
+    Parameters
+    ----------
+    movie: np.ndarray
+        The movie that is being dewarped.
+    aL: float
+        Information about the rig used for the experiment.
+    aR: float
+        Information about the rig used for the experiment.
+    bL: float
+        Information about the rig used for the experiment.
+    bR: float
+        Information about the rig used for the experiment.
+    noise_reduction: int
+        The noise reduction method that will be used in dewarping.
+    Returns
+    -------
+    dict
+        Various computed information about the video.
+    """
+    # assume flat area aL - (512-aR)
+
+    meanimg = np.mean(movie, axis=0)  # use avgimg to compute mode
+    meanimg = meanimg/256          # less noisy to get mode from 8bit histogram
+    meanimg = meanimg.astype(int)  # less noisy to get mode from 8bit histogram
```

Comment:
this isn't an 8-bit integer. It is a 64-bit integer forced into 0-256. should it be 8 bit?

Reply:
Yeah all of these comments about this are from the original code, I wasn't sure if they should stay or not.

Reply:
I don't think it actually has to be an 8bit integer. Especially since it never was, and it seemed to always work fine.

Reply:
I understand `it always worked fine` but we'd like the code not to be specific in comments like `this is an 8-bit something` when the comment is incorrect. Please at least note in the next ticket that code comments need to be reviewed.